internal/gatewayapi/securitypolicy.go

```diff
@@ -447,19 +453,58 @@ func (t *Translator) buildOIDC(
 		return nil, err
 	}
 
-	if err := validateTokenEndpoint(provider.TokenEndpoint); err != nil {
+	if err = validateTokenEndpoint(provider.TokenEndpoint); err != nil {
 		return nil, err
 	}
 	scopes := appendOpenidScopeIfNotExist(oidc.Scopes)
 
+	var (
+		redirectURL  = defaultRedirectURL
+		redirectPath = defaultRedirectPath
+		logoutPath   = defaultLogoutPath
+	)
+
+	if oidc.RedirectURL != nil {
+		path, err := extractRedirectPath(*oidc.RedirectURL)
+		if err != nil {
+			return nil, err
+		}
+		redirectURL = *oidc.RedirectURL
+		redirectPath = path
+		logoutPath = *oidc.LogoutPath
+	}
+
 	return &ir.OIDC{
 		Provider:     *provider,
 		ClientID:     oidc.ClientID,
 		ClientSecret: clientSecretBytes,
 		Scopes:       scopes,
+		RedirectURL:  redirectURL,
+		RedirectPath: redirectPath,
+		LogoutPath:   logoutPath,
 	}, nil
 }
 
+func extractRedirectPath(redirectURL string) (string, error) {
```

Comment:
can we use https://pkg.go.dev/net/url#URL instead ?

Reply:
We need to handle Envoy header values like `%REQ(x-forwarded-proto)%://%REQ(:authority)%/oauth2/callback`